### ISSUE TITLE
[Feature Request] Autofill for SignInForm and SignUpForm

### ISSUE THREAD

**Issue by fengkaijia** [aws-amplify/amplify-flutter#2189]
### Description

As a user, when login in to an app on an Amplify Authenticator sign-in form, I want to have my AutoFill service to fill in username and password for me.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [X] Authenticator
- [ ] DataStore
- [ ] Storage

### Platforms

- [ ] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Environment

```bash
N/A
```

### Dependencies

```bash
N/A
```

### Device

N/A

### OS

N/A

### Deployment Method

Amplify CLI

### Amplify Config

N/A

**Comment by Lorenzohidalgo** [comment]
@fjnoyp & @salihgueler I'll take a look into this 😉

Just as a heads-up, this change will also require the user to make some extra changes/steps outside of the application. The user will need to create, configure and host a file under that application's domain. Steps similar to when you want to [configure deeplinking](https://towardsdev.com/deeplinking-in-flutter-d09ad977b816) into your app

**Comment by Jordan-Nelson** [comment]
@Lorenzohidalgo - You are referring to setting up an [Associated Domain](https://developer.apple.com/documentation/xcode/supporting-associated-domains), correct? I think this would be optional, as it would only be required to sync auto fill between a web and mobile app. This is something that could be included in documentation for the authenticator. Let me know if you are referring to something other than Associated Domains.

**Comment by Lorenzohidalgo** [comment]
Hi @Jordan-Nelson, exactly! The official documentation mentions that it might be needed for a good iOS experience.

**Comment by Jordan-Nelson** [comment]
This auto closed when #2306 merged. I am going to re-open this. With the changes in #2306, auto fill support has been added for most form fields. I am going to leave this open to track support for auto fill groups for username password. I will update the description to reflect this.

**Comment by Lorenzohidalgo** [comment]
Hey @lucasoares have you followed the steps mentioned in the readme file?

**Comment by lucasoares** [comment]
> Hey @lucasoares have you followed the steps mentioned in the readme file? https://github.com/aws-amplify/amplify-flutter/tree/main/packages/authenticator/amplify_authenticator#setting-up--troubleshooting-autofill

The step mentioned there is only different for the iOS.. For the web, for example, it should not require any additional step, right?

**Comment by Lorenzohidalgo** [comment]
Not sure about that, I've been "away" from Flutter for a while now. My recommendation would be to follow the troubleshooting tips/guide on the referenced docs. https://api.flutter.dev/flutter/material/TextField/autofillHints.html

**Comment by lucasoares** [comment]
Maybe it's because I'm using custom builder with the custom forms to build my screen? I will wait for another member of amplify-flutter before starting to debug something from a third party OS code that should be working haha Thanks.

**Comment by lucasoares** [comment]
@Jordan-Nelson To run locally  I'm using a web-server device and accessing it in a logged-in Chrome browser... But I also tested it in my deployed web application accessing it directly by its domain.

I tried both 1password and Chrome's default password manager but there is no prompt for autofill:

![image](https://github.com/aws-amplify/amplify-flutter/assets/10624972/69af8400-bd3e-4a47-9854-6675f5849867)


My main widget:

```
  @override
  Widget build(BuildContext context) {
    return Authenticator(
      stringResolver: stringResolver,
      authenticatorBuilder: getAuthenticationBuilder(),
      child: MaterialApp.router(
         ...
      ),
    );
  }
```

My signIn form created by the `getAuthenticationBuilder`:

```
   ...
    case AuthenticatorStep.signIn:
      child = Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          _logo(),
          _logoDivider(context.l10n.signInDescribed),
          SignInForm(),
          const Spacer(),
          _dontHaveAccount(context, state),
        ],
      );
      break;
  ...
  return Center(
    child: SingleChildScrollView(
      child: ConstrainedBox(
        constraints: BoxConstraints(
          maxWidth: ContainerConstants.mediumWidth,
          maxHeight: ContainerConstants.mediumHeight,
        ),
        child: Padding(
          padding: EdgeInsets.symmetric(vertical: containerPadding),
          child: Container(
            decoration: BoxDecoration(
              borderRadius: BorderRadius.circular(10),
              border: Border.all(),
            ),
            child: Padding(
              padding: padding,
              child: child,
            ),
          ),
        ),
      ),
    ),
  );
```

**Comment by simon1389** [comment]
any updates on that?

i switched from cognito hosted ui to amplify-authenticator in my angular application. Login and Signup works fine, but passwords are not stored/autofilled as it was with the hosted ui.
It's bit of a bad user experience currently

**Comment by Jordan-Nelson** [comment]
@simon1389 - Are you using Amplify JS or Amplify Flutter?

**Comment by simon1389** [comment]
> @simon1389 - Are you using Amplify JS or Amplify Flutter?

Amplify JS

**Comment by simon1389** [comment]
> Can you open an issue at https://github.com/aws-amplify/amplify-ui? aws-amplify/amplify-ui is the repository for the Amplify JS UI components.

i created an issue, and with help of @jacoblogan we found the root cause :)

https://github.com/aws-amplify/amplify-ui/issues/5547

**Comment by Jordan-Nelson** [comment]
Great, thanks. We will continue to use this issue to track support for the **Amplify Flutter** authenticator.

**Comment by Poloten** [comment]
Are there any changes? On Android, it doesn't suggest the existing account, even though it saves the password after entering it.

**Comment by tyllark** [comment]
Hello @Poloten, could you please open a new issue so we can track it separately from this feature request. I was unable to reproduce this issue with Google Password Manager as I was able to save a new username/password and autofill the respective text fields. Autofill needs to occur 1 field at a time as we do not support [AutofillGroup](https://api.flutter.dev/flutter/widgets/AutofillGroup-class.html) currently.
